execution/src/transaction.rs

```diff
@@ -508,7 +508,16 @@ impl TransactionData {
                     // transaction as a base token transfer between EOAs.
                     Ok(Self::EoaBaseTokenTransfer(to))
                 } else {
-                    let tx_data: SerializableTransactionData = bcs::from_bytes(&tx.data)?;
+                    let tx_data: SerializableTransactionData =
+                        bcs::from_bytes(&tx.data).or_else(|e| {
```

Comment:
I think we should leave a short comment saying if it's not BCS deserializable it could be a direct ERC20 contract interaction.